core/src/consensus.rs

```diff
@@ -418,9 +489,21 @@ impl Tower {
         epoch_vote_accounts: &HashMap<Pubkey, (u64, Account)>,
     ) -> SwitchForkDecision {
         let root = self.lockouts.root_slot.unwrap_or(0);
+
         self.last_voted_slot()
             .map(|last_voted_slot| {
-                let last_vote_ancestors = ancestors.get(&last_voted_slot).unwrap();
+                let last_vote_ancestors =
+                    ancestors.get(&last_voted_slot).unwrap_or_else(|| {
+                        if !self.is_stray_last_vote() {
+                            panic!("no ancestors found with slot: {}", last_voted_slot);
```

Comment:
Double checking my assumption here 😄 

A situation like this is impossible by the time we pass the `if !self.is_stray_last_vote()` check right?

```
0 (last vote) --- 2 (snapshot)
```
Because in the above case ^ `self.last_voted_slot()` should have been empty, and we should never reach here. If this is correct, could add a comment here explaining that? Thanks!

Reply:
yeah, you're right: https://github.com/solana-labs/solana/pull/10718/commits/ada39c1c0b112d07adae6a27cc64498495a25257

Also, this prompted renewed investigation around these slot book-keeping, and I made the checked even stricter and more consistent: https://github.com/solana-labs/solana/pull/10718/commits/95ba740a1d4ff96e24272f81c7f6523a7f032cba